### PR TITLE
Switch to CSL for utxo min ada value calculation

### PR DESCRIPTION
**Pull request by errfrom**
Since now we need to know the size of the serialized transaction output in order to calculate utxo min ada value, it seems reasonable to delegate this calculation to CSL. Also, I see no reason to maintain separate logic on the CTL side. 

### Pre-review checklist

- [x] All code has been formatted using our config (`make format` for Purescript, `nixpkgs-fmt` for Nix)
- [x] All Purescript imports are explicit, including constructors
- [x] **All** existing examples have been run locally against a fully-synced testnet node
- [x] The integration and unit tests have been run (`npm run test`) locally
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`)

### CODE REVIEW

**Review by ngua — COMMENTED** [review]
Looks good, except for the thing with the changelog

**Comment by klntsky** [comment]
Could you please put more context in this PR? Especially the reason why it was decided to switch to CSL. The code is good.

File: CHANGELOG.md [hunk 1]
```diff
@@ -64,14 +64,14 @@ This release adds support for running CTL contracts against Babbage-era nodes. *
 - Switched to Ogmios for execution units evaluation ([#665](https://github.com/Plutonomicon/cardano-transaction-lib/pull/665))
 - Changed `inputs` inside `TxBody` to be `Set TransactionInput` instead `Array TransactionInput`. This guarantees ordering of inputs inline with Cardano ([#641](https://github.com/Plutonomicon/cardano-transaction-lib/pull/661))
 - Upgraded to Ogmios v5.5.0
-- Change `inputs` inside `TxBody` to be `Set TransactionInput` instead `Array TransactionInput`. This guarantees ordering of inputs inline with Cardano ([#641](https://github.com/Plutonomicon/cardano-transaction-lib/pull/661)).
+- Switched to CSL for utxo min ada value calculation ([#715](https://github.com/Plutonomicon/cardano-transaction-lib/pull/715))
```
Comment (klntsky) [inline CHANGELOG.md:67]: This should go to unreleased section.

File: CHANGELOG.md [hunk 2]
```diff
@@ -64,14 +64,14 @@ This release adds support for running CTL contracts against Babbage-era nodes. *
...
 
 ### Fixed
 
 - Handling of invalid UTF8 byte sequences in the Aeson instance for `TokenName`
 - `Types.ScriptLookups.require` function naming caused problems with WebPack ([#593](https://github.com/Plutonomicon/cardano-transaction-lib/pull/593))
 - Bad logging in `queryDispatch` that didn't propagate error messages ([#615](https://github.com/Plutonomicon/cardano-transaction-lib/pull/615))
-- UTxO Min-Ada-Value calculation ([#611](https://github.com/Plutonomicon/cardano-transaction-lib/pull/611))
+- Utxo min ada value calculation ([#611](https://github.com/Plutonomicon/cardano-transaction-lib/pull/611))
```
Comment (klntsky) [inline CHANGELOG.md:74]: Same

**Review by klntsky — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (errfrom) [reply CHANGELOG.md:78]: This must have been included in the last release

**Review by ngua — APPROVED** [review]
LGTM

**Pull request merged**